include/unifex/type_list.hpp

```diff
@@ -33,6 +33,36 @@ namespace unifex
     using apply = F<Ts...>;
   };
 
+  // apply_to_type_list<F, type_list<Ts...>>
+  //
+  // Shortcut to `typename type_list<Ts...>::template apply<F>`.
+  template <template <typename...> class F, typename List>
+  struct apply_to_type_list;
+
+  template <template <typename...> class F, typename... Ts>
+  struct apply_to_type_list<F, type_list<Ts...>> {
+    using type = typename type_list<Ts...>::template apply<F>;
+  };
+
+  template <template <typename...> class F, typename List>
+  using apply_to_type_list_t = typename apply_to_type_list<F, List>::type;
```

Comment:
Or alternatively, this could be simplified to just:
```
template <template <typename...> class F, typename List>
using apply_to_type_list_t = typename List::template apply<F>;
```

Reply:
I decided against this option though, as I decided to enforce that `List` is a `type_list<Ts...>`.